File: backend/app/schemas/notification.py
"""
Notification schemas
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime


class NotificationBase(BaseModel):
    """Base notification schema"""
    type: str
    message: str
    meta_data: Optional[Dict[str, Any]] = Field(default_factory=dict)


class NotificationCreate(NotificationBase):
    """Notification creation schema"""
    user_id: str


class NotificationResponse(NotificationBase):
    """Notification response schema"""
    id: str
    user_id: str
    read: bool
    created_at: datetime
    
    class Config:
        from_attributes = True


class NotificationListResponse(BaseModel):
    """Schema for notification list response"""
    success: bool = True
    data: dict = Field(default_factory=dict)
    message: Optional[str] = None

File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  user_id: string;
  type: string;
  message: string;
  read: boolean;
  meta_data?: Record<string, any>;
  created_at: string;
}

export interface NotificationListResponse {
  success: boolean;
  data: {
    notifications: Notification[];
    unread_count: number;
    pagination: {
      page: number;
      limit: number;
      total: number;
      pages: number;
    };
  };
}

export interface UnreadCountResponse {
  success: boolean;
  data: {
    unread_count: number;
  };
}

